lib/common/schemas.c

```diff
@@ -177,63 +195,61 @@ add_schema(enum schema_validator_e validator, const schema_version_t *version,
            const char *transform_enter, bool transform_onleave,
            int after_transform)
 {
+    struct schema_s *schema = NULL;
     int last = xml_schema_max;
     bool have_version = false;
 
     xml_schema_max++;
-    known_schemas = pcmk__realloc(known_schemas,
-                                  xml_schema_max * sizeof(struct schema_s));
-    CRM_ASSERT(known_schemas != NULL);
-    memset(known_schemas+last, 0, sizeof(struct schema_s));
-    known_schemas[last].validator = validator;
-    known_schemas[last].after_transform = after_transform;
 
-    known_schemas[last].version.v[0] = version->v[0];
-    known_schemas[last].version.v[1] = version->v[1];
+    schema = calloc(1, sizeof(struct schema_s));
+    CRM_ASSERT(schema != NULL);
+
+    schema->validator = validator;
+    schema->after_transform = after_transform;
+    schema->version.v[0] = version->v[0];
+    schema->version.v[1] = version->v[1];
 
     if (version->v[0] || version->v[1]) {
         have_version = true;
     }
 
     if (have_version) {
-        known_schemas[last].name = schema_strdup_printf("pacemaker-", *version, "");
+        schema->name = schema_strdup_printf("pacemaker-", *version, "");
     } else {
         CRM_ASSERT(name != NULL);
-        schema_scanf(name, "%*[^-]-", known_schemas[last].version, "");
-        known_schemas[last].name = strdup(name);
-        CRM_ASSERT(known_schemas[last].name != NULL);
+        schema_scanf(name, "%*[^-]-", schema->version, "");
```

Comment:
CI is failing because coverity is complaining about not checking scanf here.  This isn't new - it's just coming up now because the line changed.  It's not a problem to check it, but I'm not really sure what we'd do with the result.

Reply:
Added a patch on top to remove the unnecessary code here.